Plugins docs (#21826)
grafana/grafana: docs/sources/plugins/developing/development.md

> 1. [Setup grafana](http://docs.grafana.org/project/building_from_source/)

Reviewer: ```suggestion
1. [Set up Grafana](https://github.com/grafana/grafana/blob/master/contribute/developer-guide.md)
```

2. Clone an example plugin into ```/var/lib/grafana/plugins```  or `data/plugins` (relative to grafana git repo if you're running development version from source dir)

> 3. Use one of our example plugins as a starting point

Reviewer: @marcusolsson , is this still true?

Reviewer 2: yes -- i mean there is no better option

- [Clock panel](https://github.com/grafana/clock-panel)
- [Pie chart panel](https://github.com/grafana/piechart-panel)

> There are two blog posts about authoring a plugin that might also be of interest to other users.

Reviewer: See earlier comment about audience.

Also, is this content written for Grafana 3.0 still valid for 6.6?

Author: @oddlittlebird I was told we shouldn't point to old blog posts but instead to our recently released tutorials section.

Reviewer: @marcusolsson , are the tutorials ready?

Author: @oddlittlebird https://grafana.com/tutorials/

- [Timing is Everything. Writing the Clock Panel Plugin for Grafana](https://grafana.com/blog/2016/04/08/timing-is-everything.-writing-the-clock-panel-plugin-for-grafana-3.0/)
- [Timing is Everything. Editor Mode in Grafana for the Clock Panel Plugin](https://grafana.com/blog/2016/04/15/timing-is-everything.-editor-mode-in-grafana-3.0-for-the-clock-panel-plugin/).

## What languages?

> Since everything turns into javascript it's up to you to choose which language you want. That said, it's probably a good idea to choose es6 or typescript since we use es6 classes in Grafana. So it's easier to get inspiration from the Grafana repo if you choose one of those languages.

Reviewer: ```suggestion
Since everything turns into JavaScript, it's up to you to choose which language you want. That said, it's probably a good idea to choose es6 or TypeScript, because we use es6 classes in Grafana. So it's easier to get inspiration from the Grafana repo if you choose one of those languages.
```

## Buildscript

> You can use any build system you like that supports systemjs. All the built content should end up in a folder named ```dist``` and committed to the repository. By committing the dist folder the person who installs your plugin does not have to run any build script. All of our example plugins have build script configured.

Reviewer: ```suggestion
You can use any build system that supports systemjs. All the built content should end up in a folder named ```dist``` and be committed to the repository. By committing the dist folder, the person who installs your plugin does not have to run any build script. All of our example plugins have a build script configured.
```

Reviewer: Do we want to mention that a README in the dist folder overrides the README on the Plugins website? @DanCech ?

There are three ways that you can start developing a Grafana plugin.

> 1. Setup a Grafana development environment [(described here)](http://docs.grafana.org/project/building_from_source/) and place your plugin in the ```data/plugins``` folder.

Reviewer: ```suggestion
1. Set up a Grafana development environment as described in the Grafana [Developer guide](https://github.com/grafana/grafana/blob/master/contribute/developer-guide.md) and place your plugin in the ```data/plugins``` folder.
```

2. Install Grafana and place your plugin in the plugins directory which is set in your [config file](/docs/grafana/latest/installation/configuration). By default, this is `/var/lib/grafana/plugins` on Linux systems.
3. Place your plugin directory anywhere you like and specify it in grafana.ini.

> We encourage people to setup the full a Grafana environment so that you can get inspiration from the rest of grafana code base.

Reviewer: ```suggestion
We encourage people to set up the full Grafana environment so that you can get inspiration from the rest of the Grafana code base.
```

Reviewer: Watch out for "setup." "Setup" is a noun. "Set up" is a verb.

> When Grafana starts, it will scan the plugin folders and mount every folder that contains a plugin.json file unless

Reviewer: ```suggestion
When Grafana starts, it scans the plugin folders and mounts every folder that contains a plugin.json file unless
```

> the folder contains a subfolder named dist. In that case grafana will mount the dist folder instead.

Reviewer: ```suggestion
the folder contains a subfolder named dist. In that case, Grafana mounts the dist folder instead.
```

> This makes it possible to have both built and src content in the same plugin git repo.

Reviewer: ```suggestion
This makes it possible to have both built and src content in the same plugin Git repo.
```

> We currently have three different examples that you can fork/download to get started developing your grafana plugin.

Reviewer: ```suggestion
We have three different examples that you can fork/download to get started developing your Grafana plugin.
```

Reviewer: @marcusolsson and @mckn , I suspect you will want to add or replace this section around the May timeframe.
